### PR TITLE
SQ-55/About-to-start notification creation

### PR DESCRIPTION
**Pull request by danybony**
### Description
This PR adds the logic to create and display notification for event(s) about to start.
It does not include the logic to GET which events are about to start (we are still missing the time slot for that).
I also added a debug activity to test these notifications with mock events.
The information we are displaying is:
* Event title
* Event room
* Event Speakers (only for single event)

### Screenshots
| Single | Multiple < 7.0 | Multiple >= 7.0 |
| --- | --- | --- |
| device-2017-02-13-231839 | device-2017-02-13-231821 | device-2017-02-13-232135 |

### CODE REVIEW

File: app/src/main/java/net/squanchy/notification/NotificationCreator.java [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+        Intent baseIntent = new Intent(context, ScheduleActivity.class);
+        baseIntent.setFlags(Intent.FLAG_ACTIVITY_NEW_TASK | Intent.FLAG_ACTIVITY_CLEAR_TASK);
+        return TaskStackBuilder.create(context)
+                .addNextIntent(baseIntent);
+    }
+
+    private NotificationCompat.BigTextStyle createBigTextRichNotification(NotificationCompat.Builder notificationBuilder, Event event) {
+        StringBuilder bigTextBuilder = new StringBuilder()
+                .append(getDisplayedSpeakers(event));
+        if (event.place() != null) {
```
Comment (danybony) [inline app/src/main/java/net/squanchy/notification/NotificationCreator.java:123]: I see that we are doing this check elsewhere in the codebase. Is this really necessary?
If we can assume the place will always be there I'd be happy to remove these checks

File: app/build.gradle [hunk 2]
```diff
@@ -77,6 +77,9 @@ android {
         }
 
         buildTypes {
+            debug {
+                minifyEnabled false
```
Comment (rock3r) [inline app/build.gradle:81]: I thought this was the default

File: app/src/main/java/net/squanchy/notification/NotificationCreator.java (hunk 1 above)
Comment (rock3r) [reply app/src/main/java/net/squanchy/notification/NotificationCreator.java:123]: Not sure. I think we need to redefine our domain models (the ones you're using are viewmodels), maybe we can sync up with @fourlastor and @tiwiz on this?

File: app/src/main/java/net/squanchy/eventdetails/EventDetailsService.java [hunk 3]
```diff
@@ -44,7 +45,7 @@
                     apiEvent.name,
                     apiEvent.place,
                     apiEvent.experienceLevel,
-                    map(speakers, toSpeakerName())
+                    map(speakers, toSpeakers())
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/eventdetails/EventDetailsService.java:48]: Thanks for this, we needed to do it at some point, BUT we need to create new view models to include the data we actually need in the screens

**Review by rock3r — APPROVED** [review]
🚢

**Pull request merged**